File: python/python/ch5/Wrapper.py
class Wrapper:
    def __init__(self, value):
        self._value = value
        
    def map(self, f):
        return f(self._value)
      
    def fmap(self, f):
        return Wrapper(f(self._value))
      
    def to_string(self):
        return f'Wrapper({self._value})'

File: javascript/ch5/Wrapper.js
function Wrapper(value) {
  const _value = value;

  return {
    map(f) {
      return f(_value);
    },

    fmap(f) {
      return Wrapper(f(_value));
    },

    toString() {
      return `Wrapper(${_value})`;
    }
  };
}

module.exports = {
  Wrapper
};